### PR TITLE
Update ForceMarshalBroker to not require runtime type match

### PR DESCRIPTION
**Pull request by BertanAygun**
ForceMarshal broker was calling inner GetProxyAsync which required a runtime type match between service and proxy requested. This is not an actual requirement when using remote service brokers.

This PR updates the behavior to be match more to the remote scenario.

### CODE REVIEW

**Comment by BertanAygun** [comment]
Tagging @AArnott as I realized I can't add a reviewer once I publish the PR.

File: src/Microsoft.ServiceHub.Framework/ServiceBrokerAggregator.cs [hunk 1]
```diff
@@ -221,15 +221,22 @@ public ForceMarshalingBroker(IServiceBroker inner)
 		public async ValueTask<T?> GetProxyAsync<T>(ServiceRpcDescriptor serviceDescriptor, ServiceActivationOptions options, CancellationToken cancellationToken)
 			where T : class
 		{
-			T? client = await this.inner.GetProxyAsync<T>(serviceDescriptor, options, cancellationToken).ConfigureAwait(false);
-			if (client == null)
+			IDuplexPipe? pipe = await this.inner.GetPipeAsync(serviceDescriptor.Moniker, options, cancellationToken).ConfigureAwait(false); ;
```
Comment (AArnott) [inline src/Microsoft.ServiceHub.Framework/ServiceBrokerAggregator.cs:224]: you have double-semicolons at the end of this line.

File: test/Microsoft.ServiceHub.Framework.Tests/ServiceBrokerAggregatorTests.cs [hunk 2]
```diff
@@ -192,7 +198,7 @@ public async Task ForceMarshal_GetProxyAsync()
 		ICalculator? result = await aggregator.GetProxyAsync<ICalculator>(Descriptor1, this.TimeoutToken);
 		Assumes.NotNull(result);
 		Assert.NotSame(result, this.mockBrokers[1].PipeResults[Descriptor1.Moniker]);
-		Assert.Equal(8, await result.AddAsync(3, 5));
```
Comment (AArnott) [inline test/Microsoft.ServiceHub.Framework.Tests/ServiceBrokerAggregatorTests.cs:195]: Why did you need to remove the AddAsync line?

**Review by AArnott — CHANGES_REQUESTED** [review]
Thanks. This looks like a good change.

File: test/Microsoft.ServiceHub.Framework.Tests/ServiceBrokerAggregatorTests.cs (hunk 2 above)
Comment (BertanAygun) [reply test/Microsoft.ServiceHub.Framework.Tests/ServiceBrokerAggregatorTests.cs:195]: The MockPipe returned from GetPipeAsync method of MockBroker in these tests don't actually do anything. I assumed that's why GetPipeAsync tests wasn't calling any methods either.

File: src/Microsoft.ServiceHub.Framework/ServiceBrokerAggregator.cs (hunk 1 above)
Comment (BertanAygun) [reply src/Microsoft.ServiceHub.Framework/ServiceBrokerAggregator.cs:224]: I had warnings hidden in VS for an earlier solution so didn't realize it. Fixed in the last iteration.

**Review by AArnott — APPROVED** [review]
I'm not thrilled if the ForceMarshal method doesn't have any tests that actually exercise the pipes it sets up, but as filling this looks like some substantial work, I'll leave it up to you.

**Review by AArnott — APPROVED** [review]
Awesome. Thanks.
